plugins/Compressor/Compressor.cpp

```diff
@@ -0,0 +1,642 @@
+					: m_attCoeff;
+
+				m_yL[i] = m_yL[i] * att + (1 - att) * t;
+				m_holdTimer[i] = m_holdLength;// Reset hold timer
+			}
+			else// Release phase
+			{
+				// Calculate release value depending on crest factor
+				const float rel = m_autoRelVal
+					? msToCoeff(2.f * m_compressorControls.m_releaseModel.value() / ((m_crestFactorVal[i] - 1) * m_autoRelVal + 1))
+					: m_relCoeff;
+
+				if (m_holdTimer[i])// Don't change peak if hold is being applied
+				{
+					--m_holdTimer[i];
+				}
+				else
+				{
+					m_yL[i] = m_yL[i] * rel + (1 - rel) * t;
+				}
+			}
+
+			// Keep it above the noise floor
+			m_yL[i] = qMax(COMP_NOISE_FLOOR, m_yL[i]);
+
+			// For the visualizer
+			m_displayPeak[i] = m_yL[i];
+
+			const float currentPeakDbfs = ampToDbfs(m_yL[i]);
+
+			// Now find the gain change that should be applied,
+			// depending on the measured input value.
+			if (currentPeakDbfs - m_thresholdVal < -m_kneeVal)// Below knee
+			{
+				gainResult[i] = currentPeakDbfs;
+			}
+			else if (abs(currentPeakDbfs - m_thresholdVal) < m_kneeVal)// Within knee
+			{
+				const float temp = currentPeakDbfs - m_thresholdVal + m_kneeVal;
+				gainResult[i] = currentPeakDbfs + ((limiter ? 0 : m_ratioVal) - 1) * temp * temp / (4 * m_kneeVal);
+			}
+			else// Above knee
+			{
+				gainResult[i] = limiter
+					? m_thresholdVal
+					: (m_thresholdVal + (currentPeakDbfs - m_thresholdVal) * m_ratioVal);
+			}
+
+			gainResult[i] = dbfsToAmp(gainResult[i]) / m_yL[i];
+			gainResult[i] = qMax(m_rangeVal, gainResult[i]);
+		}
+
+		switch (stereoLink)
+		{
+			case 1:// Maximum
+			{
+				gainResult[0] = gainResult[1] = qMin(gainResult[0], gainResult[1]);
+				break;
+			}
+			case 2:// Average
+			{
+				gainResult[0] = gainResult[1] = (gainResult[0] + gainResult[1]) * 0.5f;
+				break;
+			}
+			case 3:// Minimum
+			{
+				gainResult[0] = gainResult[1] = qMax(gainResult[0], gainResult[1]);
+				break;
+			}
+			case 4:// Blend
+			{
+				if (blend > 0)// 0 is unlinked
+				{
+					if (blend <= 1)// Blend to minimum volume
+					{
+						const float temp1 = qMin(gainResult[0], gainResult[1]);
+						gainResult[0] = linearInterpolate(gainResult[0], temp1, blend);
+						gainResult[1] = linearInterpolate(gainResult[1], temp1, blend);
+					}
+					else if (blend <= 2)// Blend to average volume
+					{
+						const float temp1 = qMin(gainResult[0], gainResult[1]);
+						const float temp2 = (gainResult[0] + gainResult[1]) * 0.5f;
+						gainResult[0] = linearInterpolate(temp1, temp2, blend - 1);
+						gainResult[1] = gainResult[0];
+					}
+					else// Blend to maximum volume
+					{
+						const float temp1 = (gainResult[0] + gainResult[1]) * 0.5f;
+						const float temp2 = qMax(gainResult[0], gainResult[1]);
+						gainResult[0] = linearInterpolate(temp1, temp2, blend - 2);
+						gainResult[1] = gainResult[0];
+					}
+				}
+				break;
+			}
+		}
+
+		// Bias compression to the left or right (or mid or side)
+		if (stereoBalance != 0)
```

Comment:
You avoided comparing to 0 elsewhere. I think you should do the same here.

Reply:
The other areas are boolean.  `stereoBalance` is a float (from -1 to 1), so I think it would be best to explicitly compare it to zero.